runtime/src/impls.rs

```diff
@@ -223,6 +225,44 @@ impl<F: FindAuthor<u32>> FindAuthor<H160> for EthereumFindAuthor<F> {
 	}
 }
 
+/// Implements the transaction payment for a pallet implementing the `Currency`
+/// trait (eg. the pallet_balances) using an unbalance handler (implementing
+/// `OnUnbalanced`).
+/// Similar to `CurrencyAdapter` of `pallet_transaction_payment`
+pub struct CENNZnetEVMCurrencyAdapter<T, C>(PhantomData<(T, C)>);
+
+impl<T, C> OnChargeEVMTransaction<T> for CENNZnetEVMCurrencyAdapter<T, C>
+where
+	T: pallet_evm::Config + crml_staking::Config,
+	C: Currency<<T as frame_system::Config>::AccountId>,
+	C::PositiveImbalance:
+		Imbalance<<C as Currency<<T as frame_system::Config>::AccountId>>::Balance, Opposite = C::NegativeImbalance>,
+	C::NegativeImbalance:
+		Imbalance<<C as Currency<<T as frame_system::Config>::AccountId>>::Balance, Opposite = C::PositiveImbalance>,
+{
+	type LiquidityInfo = Option<
+		<<T as pallet_evm::Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance,
+	>;
+
+	fn withdraw_fee(who: &H160, fee: U256) -> Result<Self::LiquidityInfo, pallet_evm::Error<T>> {
+		<() as OnChargeEVMTransaction<T>>::withdraw_fee(who, fee)
+	}
+
+	fn correct_and_deposit_fee(who: &H160, corrected_fee: U256, already_withdrawn: Self::LiquidityInfo) {
+		<() as OnChargeEVMTransaction<T>>::correct_and_deposit_fee(who, corrected_fee, already_withdrawn)
+	}
+
+	fn pay_priority_fee(tip: U256) {
+		let controller = T::AddressMapping::into_account_id(<pallet_evm::Pallet<T>>::find_author());
```

Comment:
this address will not be the controller. The result of this function is going to get the controller convert to 20 byte then to a new address altogether `c -> e -> c'`

some ideas to solve this are to add a function to rewards pallet so `c` can withdraw from `c'` but the manual withdraws aren't the best

Reply:
Ah true, that makes it complicated. I'll have a think on some solutions

Reply:
to give some context: `pallet_evm::Pallet<T>>::find_author()` should return a 20 byte address that can be uniquely tied to the validator. 

contracts or EVM tooling will treat this address as the 'miner'
so it will show up in evm block info and when a contract transfers to `block.coinbase`it will also pay this address.